lessons/babel.md

```diff
@@ -1 +1,130 @@
+In this lesson we'll be learning about Babel which allows us to write modern javascript without worrying about it!
+
+We'll go through briefly what Babel does, why we use it and how to set it up in a project.
+
+When opening the project and looking around, if you haven't looked at JS for a while there may be some newer syntax you don't understand. It's not just you; some older browsers won't understand this either, which is why we want Babel.
+### What is Babel?
+- It is a 'transpiler', converting Javascript to Javascript
+
+### Why do we want this
+- Javascript has many different runtimes
+  - V8 (Chrome)
+  - Chakra (IE >= 9)
+  - Spidermonkey (Firefox)
+  - JavaScriptCore (Safari)
+  - JScript (IE < 9)
+- Some of the newer language features are unsupported in many of the currently used browsers
+- We want to be able to use modern features that make Javascript easier (and more fun) to develop in
+- Almost every browser that hits the Guardian supports the 'ES5' spec, Babel converts features from later specs back to ES5
+
+### What features does Babel handle?
+- Many of the improvements in the newer specs are syntatic sugar, which make transpilation relatively easy (default parameters, implicit returns)
+```js
+// ES2015
+const myFunc = (a = 1) => a + 2;
+
+// ES5 equivalent
+function myFunc(a) {
+  a = typeof a === "undefined" ? 1 : a;
+  return a + 2;
+}
+```
+- Some of the language features are new native APIs that are implementable (albeit less efficiently) in ES5 javascript and require a polyfill (e.g. Promise), this isn't handled by Babel transpilation! (Guardian uses polyfill.io [does it?!])
+- Some are not possible to completely polyfill (e.g. extending a built in class isn't possible in ES5) so they approximate functionality but may not work in all cases.
+
+## The project
+The app we're looking at revolves around a *tiny* Virtual DOM library (a pattern popular with libraries like React and Vue).
+
+The app make a request to CAPI and then uses the VDOM library to render the response the DOM and show a "Guardian" front page.
+
+We can test out the project now by running `yarn start` inside the project folder and navigating to the URL shown in the output. This will probably work fine in Chrome and Firefox but in Safari (at time of writing) it will have trouble with some of the syntax.
+
+## Getting Babel into a project
+This is the process to follow to get Babel into our project:
+- Install babel and it's CLI tool into the project
+- Configure babel: tell it what syntax we want to transform
+- Run babel on our code
+- Add in a polyfill for any missing APIs (babel doesn't handle these)
+
+### Installing Babel
+The Babel docs have pretty comprehensive info on this but it can sometimes be a lot to get your head around for the first time.
+
+- So first let's install babel into our project using our package manager
+  - We're using `yarn` so we need to run `yarn add --dev @babel/core @babel/cli` (for `npm i --save-dev @babel/core @babel/cli`)
```

Comment:
👍 for using beta – it's so imminent, this is 👌

Reply:
Exactly, I thought it'd just fall out of date almost by the time we ran the lesson!